src/backend/graphics/gl.rs

```diff
@@ -0,0 +1,55 @@
+//! Gl rendering types
+
+use nix::libc::c_void;
+
+use super::{PixelFormat, SwapBuffersError};
+
+#[cfg_attr(feature = "cargo-clippy", allow(clippy))]
+#[allow(missing_docs)]
+pub(crate) mod ffi {
+    include!(concat!(env!("OUT_DIR"), "/gl_bindings.rs"));
+}
+
+pub use self::ffi::Gles2;
+
+/// Trait that describes objects that have an OpenGL context
+/// and can be used to render upon
+pub trait GLGraphicsBackend {
+    /// Swaps buffers at the end of a frame.
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError>;
+
+    /// Returns the address of an OpenGL function.
+    ///
+    /// Supposes that the context has been made current before this function is called.
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void;
+
+    /// Returns the dimensions of the window, or screen, etc in points.
+    ///
+    /// That are the scaled pixels of the underlying graphics backend.
+    /// For nested compositors this will respect the scaling of the root compositor.
```

Comment:
What do you mean by "respect the scaling"? AFAIK the winit backend transposes back the pixels to expose the "physical pixels" (as winit calls them), is that what you have in mind? This formulation could be clarified.